airflow/providers/salesforce/operators/bulk.py

```diff
@@ -0,0 +1,113 @@
+    :param use_serial: Process batches in serial mode
+    :param salesforce_conn_id: The :ref:`Salesforce Connection id <howto/connection:SalesforceHook>`.
+    """
+
+    def __init__(
+        self,
+        *,
+        operation: Optional[str] = None,
+        object_name: Optional[str] = None,
+        payload: list,
+        external_id_field: str = 'Id',
+        batch_size: int = 10000,
+        use_serial: bool = False,
+        salesforce_conn_id: str = 'salesforce_default',
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.operation = operation
+        self.object_name = object_name
+        self.payload = payload
+        self.external_id_field = external_id_field
+        self.batch_size = batch_size
+        self.use_serial = use_serial
+        self.salesforce_conn_id = salesforce_conn_id
+        self._validate_inputs()
+
+    def _validate_inputs(self) -> None:
+        if not self.object_name:
+            raise ValueError("The required parameter 'object_name' is missing.")
+
+        available_operations = ['insert', 'update', 'upsert', 'delete', 'hard_delete']
+        if self.operation not in available_operations:
+            raise ValueError(f"Operation not found! Available operations are {available_operations}.")
+
+    def execute(self, context: 'Context') -> dict:
+        """
+        Makes an HTTP request to Salesforce Bulk API.
+
+        :param context: The task context during execution.
+        :return: API response if do_xcom_push is True
+        :rtype: dict
+        """
+        sf_hook = SalesforceHook(salesforce_conn_id=self.salesforce_conn_id)
+        conn = sf_hook.get_conn()
+
+        result: dict = {}
+        if self.operation == 'insert':
+            result = conn.bulk.__getattr__(self.object_name).insert(
+                data=self.payload, batch_size=self.batch_size, use_serial=self.use_serial
+            )
+        elif self.operation == 'update':
+            result = conn.bulk.__getattr__(self.object_name).update(
+                data=self.payload, batch_size=self.batch_size, use_serial=self.use_serial
+            )
+        elif self.operation == 'upsert':
+            result = conn.bulk.__getattr__(self.object_name).upsert(
+                data=self.payload,
+                external_id_field=self.external_id_field,
+                batch_size=self.batch_size,
+                use_serial=self.use_serial,
+            )
+        elif self.operation == 'delete':
+            result = conn.bulk.__getattr__(self.object_name).delete(
+                data=self.payload, batch_size=self.batch_size, use_serial=self.use_serial
+            )
+        elif self.operation == 'hard_delete':
+            result = conn.bulk.__getattr__(self.object_name).hard_delete(
+                data=self.payload, batch_size=self.batch_size, use_serial=self.use_serial
+            )
+
+        if not self.do_xcom_push:
+            result = None
+
+        return result
```

Comment:
```suggestion
        if self.do_xcom_push and result:
            return result

        return None
```

Reply:
@eladkal 
I referred to [this code](https://github.com/apache/airflow/blob/main/airflow/providers/salesforce/operators/salesforce_apex_rest.py#L55-L69).

I removed the return type because None cannot be returned if return type is specified.